features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Listing/ApiThumb.js

```diff
@@ -93,24 +105,30 @@ class ApiThumb extends React.Component {
                                 More...
                             </Button>
                         </Link>
-                        <Dialog open={this.state.openUserMenu} transition={Slide} onRequestClose={this.handleRequestClose}>
+                        <ScopeValidation resourcePath={resourcePath.SINGLE_API}
+                                         resourceMethod={resourceMethod.DELETE}>
+                            <Button dense color="primary" onClick={this.handleRequestOpen}>Delete</Button>
+                        </ScopeValidation>
+                        <Dialog open={this.state.openUserMenu} transition={Slide}
+                                onRequestClose={this.handleRequestClose}>
                             <DialogTitle>
-                                {"Use Google's location service?"}
+                                {"Confirm"}
                             </DialogTitle>
                             <DialogContent>
                                 <DialogContentText>
                                     Are you sure you want to delete the API ({name} - {version})?
                                 </DialogContentText>
                             </DialogContent>
                             <DialogActions>
-                                <Button onClick={this.handleRequestClose} color="primary">
+                                <Link to={"/apis/"}>
```

Comment:
Why we put the `Delete` Button inside a React-Router `Link` tag ?

Reply:
this is used in grid view, After delete API, need to render the page if not when we go to list view directly without refresh that 'deleted API' apper in list view. any way to avoid error without using this?

Reply:
Yes, IMHO we could use [state uplifting](https://reactjs.org/docs/lifting-state-up.html) method to get the parent state update via child element.